functional-tests/factory/s2/__init__.py

```diff
@@ -0,0 +1,91 @@
+import os
+from dataclasses import asdict
+from pathlib import Path
+
+import flexitest
+import toml
+
+from utils.utils import OperatorKeyInfo
+
+from .config_cfg import S2Config, TlsConfig, TransportConfig
+
+
+class S2Factory(flexitest.Factory):
+    def __init__(self, port_range: list[int]):
+        super().__init__(port_range)
+
+    @flexitest.with_ectx("ctx")
+    def create_s2_service(
+        self, name: str, operator_key: OperatorKeyInfo, ctx: flexitest.EnvContext
+    ) -> flexitest.Service:
+        datadir = ctx.make_service_dir(name)
+
+        base = Path(ctx.envdd_path)
+        mtls_cred = str((base / "../s2_cred/tls").resolve())
+
+        # Dynamic ports
+        s2_port = self.next_port()
+
+        # write seed file
+        seed_file = str((base / name / "seed").resolve())
+        write_s2_seed(seed_file, operator_key)
+
+        # write s2 config
+        config_toml = str((base / name / "config.toml").resolve())
+        generate_s2_config(config_toml, mtls_cred, seed_file, s2_port)
+
+        logfile = os.path.join(datadir, "service.log")
+
+        cmd = [
+            "secret-service",
+            config_toml,
+        ]
+
+        props = {
+            "s2_port": s2_port,
+        }
+
+        svc = flexitest.service.ProcService(props, cmd, stdout=logfile)
+        svc.start()
+        return svc
+
+
+def generate_s2_config(output_path: str, mtls_cred: str, seed_file: str, s2_port: int):
+    """
+    Generate S2 service TOML config file using dataclass configuration.
```

Comment:
i think it's fine to have concise names but it's better to have full name on comments.

```suggestion
    Generate secret service TOML config file using dataclass configuration.
```